lib/GADS/Schema/Result/Report.pm

```diff
@@ -304,12 +294,37 @@ sub create_pdf
         fg_color  => '#ffffff',
     };
 
-    $pdf->table(
-        data         => $fields,
-        header_props => $hdr_props,
-        border_c     => '#007c88',
-        h_border_w   => 1,
-    );
+    my $d=$record->columns_render;
+    my @layouts = $self->report_layouts;
+    my $result = [];
+    my %include;
+
+    for my $val (@layouts) {
+        $include{$val->layout->name} = 1;
+    }
+
+    foreach my $col (@$d) {
+        next unless $include{$col->name};
+        push @$result, $col;
+    }
```

Comment:
I think this whole section could be simplified, something like (include comments too):
```
# Columns from the report
my %include = map { $_->layout_id => 1 } $self->report_layouts;
# Filter out columns the user does not have access to
my $result = [grep $include{$_->id}, @{$reord->columns_render}];
```